### PR TITLE
phpdocumentor: format with nixfmt-rfc-style; switch to buildComposerProject2

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by piotrkwiecinski** [comment]
Result of `nixpkgs-review pr 336145` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phpdocumentor</li>
  </ul>
</details>

**Review by drupol — APPROVED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
Oops, incorrect hash.

File: pkgs/by-name/ph/phpdocumentor/package.nix [hunk 1]
```diff
@@ -15,7 +16,7 @@ php.buildComposerProject (finalAttrs: {
     hash = "sha256-zpAKygpxyKYfefa5ag76saTSQNLon/v3rYkl0Nj2+FM=";
   };
 
-  vendorHash = "sha256-VNlAzWueF7ZXBpr9RrJghMPrAUof7f1DCh1osFIwFfs=";
+  vendorHash = "sha256-PJ5D48C1JThXblMlADm7YqI1iITsK/MYZW1fpkNr+54=";
```
Comment (drupol) [inline pkgs/by-name/ph/phpdocumentor/package.nix:19]:
```suggestion
  vendorHash = "sha256-Kc88CPo+86EnnCyGW+u8hHx3qxvhGyASxwYiXlEneMc=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ph/phpdocumentor/package.nix (hunk 1 above)
Comment (piotrkwiecinski) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: I run a build with empty hash and got different value `sha256-NfG69h0QIh/uOVi8Yi274ZK+sWvpEbiUrTd/ifStA7w=` in meantime. Maybe it's non-deterministic for some reason?

Comment (drupol) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: Let me check again...

Comment (drupol) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: Now I find `sha256-PJ5D48C1JThXblMlADm7YqI1iITsK/MYZW1fpkNr+54=`. Weird. There's something unstable somewhere...

**Comment by drupol** [comment]
@ofborg build phpdocumentor

File: pkgs/by-name/ph/phpdocumentor/package.nix (hunk 1 above)
Comment (piotrkwiecinski) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: ofborg shows:
```
error: hash mismatch in fixed-output derivation '/nix/store/5qqkjhnkx2ninwpxfg20h0hy3a5nhb6z-phpdocumentor-3.5.3-composer-repository.drv':
         specified: sha256-PJ5D48C1JThXblMlADm7YqI1iITsK/MYZW1fpkNr+54=
            got:    sha256-L8QOA2O4lLf5y7oAkyB+ALaMRmwA9Cxsvw9x77OHtik=
```
and

```
error: hash mismatch in fixed-output derivation '/nix/store/flr9prqzn4lb52n31ps7zgbbq8yqjm3i-phpdocumentor-3.5.3-composer-repository.drv':
         specified: sha256-PJ5D48C1JThXblMlADm7YqI1iITsK/MYZW1fpkNr+54=
            got:    sha256-Kc88CPo+86EnnCyGW+u8hHx3qxvhGyASxwYiXlEneMc=
```

Comment (drupol) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: There's something unstable, that's what I feared. We need to find the issue now...

Comment (drupol) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: I quickly did this: 
image

But I will continue tomorrow... if you find anything in the meantime, open a PR to fix the issue in the builder.

**Comment by drupol** [comment]
It looks like something unstable in Composer when generating the file `autoload_static.php`.

image

**Comment by drupol** [comment]
I'm pretty sure this is because reading files on a filesystem is not deterministic. I guess we will need to patch Composer to make sure the paths are sorted.

File: pkgs/by-name/ph/phpdocumentor/package.nix (hunk 1 above)
Comment (piotrkwiecinski) [reply pkgs/by-name/ph/phpdocumentor/package.nix:19]: Maybe optimize autoload causes the issue

**Comment by drupol** [comment]
I would put this on hold until we find out why the new builder is unstable.

**Comment by drupol** [comment]
> It looks like something unstable in Composer when generating the file `autoload_static.php`.
> 
> image

Maybe @seldaek can help here, if not, I'll debug Composer next week.

**Comment by Seldaek** [comment]
Sorry not sure what would cause this. We do sort packages, and then we do sort namespaces.. But maybe something somewhere is missing.

**Comment by drupol** [comment]
@piotrkwiecinski Why closing this? It's true that I forgot about this PR, but I think it's OK to submit it to nixpkgs now that the version 2 of the ComposerBuilder is more stable.

**Comment by piotrkwiecinski** [comment]
@drupol it had conflicts. I'll reopen it tomorrow

**Comment by drupol** [comment]
Fair enough :)

**Pull request closed without merging**